hs/test-examples/features/default_fn_arguments_dependant.rsh

```diff
@@ -0,0 +1,14 @@
+'reach 0.1';
+
+const a = 2;
+
+export const foo = (x, y, z = (x + y + a), t = z - 2, u = x + x) =>
+  x + y + z + t + u;
+
+export const main =
+  Reach.App(
+    {},
+    [Participant('A', {})],
+    (A) => {
+      assert(foo(1, 2) == 13);
```

Comment:
Can we add a `const a = 10;` or something above this as a test that it isn't using the caller environment? (I believe the code is right, but I want that tested explicitly)